File: testData/src/java8/pkg/TestTryLoopReturnFinally.java
package pkg;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class TestTryLoopReturnFinally {
  private boolean field;

  public void test(File file) {
    try {
      while (this.field) {
        if (file == null) {
          return;
        }

        Scanner scanner = new Scanner(file);
      }
    } catch (FileNotFoundException e) {
      e.printStackTrace();
    } finally {
      System.out.println("Finally");
    }
  }
}

File: plugins/kotlin/testData/src/kt/pkg/TestTryLoopReturnFinally.kt
package pkg

import java.io.File
import java.io.FileNotFoundException
import java.util.*

class TestTryLoopReturnFinally {
    private val field = false

    fun test(file: File?) {
        try {
            while (this.field) {
                if (file == null) {
                    return
                }

                val scanner = Scanner(file)
            }
        } catch (e: FileNotFoundException) {
            e.printStackTrace()
        } finally {
            println("Finally")
        }
    }
}
